src/meta/processors/index/CreateEdgeIndexProcessor.cpp

```diff
@@ -96,7 +97,15 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
     }
 
     if (checkIndexExist(fields, item)) {
-      resp_.code_ref() = nebula::cpp2::ErrorCode::E_EXISTED;
+      if (ifNotExists) {
+        resp_.code_ref() = nebula::cpp2::ErrorCode::SUCCEEDED;
+        cpp2::ID thriftID;
+        // Fill index id to avoid broken promise
+        thriftID.index_id_ref() = item.get_index_id();
+        resp_.id_ref() = thriftID;
```

Comment:
It feels a little weird this way
1. The first `ifNotExists` in line 45 is to judge whether the index name exists or not.

2 This place is to judge whether there are other indexes that contain this fields
for example
```
create edge e(a int, b int);
create edge index e_index on e(a, b)
create edge index if not exists e_index_2 on e(a);
```
This returns a successful creation and returns the indexId of the first index.
There is actually only one index e_index

Reply:
It shows that the index creation is successful, but when show, the index name does not exist

Reply:
Indeed, this is the point just as you said. 
IMO, index name is a redundant design. 
A better syntax might be this: 
```
create edge index on e(a);
create edge index if not exists on e(a);
rebuild edge index on e(a);
```
In this way, the implementation layer does not need to verify the index name(line 45).
Considering compatibility, this pr is not going to do major refactoring.

Reply:
@czpmango @MuYiYong How is our GQL version index implemented?

Reply:
We should try to avoid similar problems in GQL version.

Reply:
5.0 index design needs to consider this problem, thx @MuYiYong